components/signin/public/identity_manager/brave_identity_manager_unittest.cc

```diff
@@ -0,0 +1,267 @@
+    std::vector<gaia::ListedAccount> signed_out_accounts;
+    bool accounts_are_fresh = gaia_cookie_manager_service()->ListAccounts(
+        &signed_in_accounts, &signed_out_accounts);
+    static_cast<void>(accounts_are_fresh);
+  }
+
+  const CoreAccountId& primary_account_id() const {
+    return primary_account_id_;
+  }
+
+  TestSigninClient* signin_client() { return &signin_client_; }
+
+  network::TestURLLoaderFactory* test_url_loader_factory() {
+    return &test_url_loader_factory_;
+  }
+
+ private:
+  base::ScopedTempDir temp_profile_dir_;
+  base::test::TaskEnvironment task_environment_;
+  sync_preferences::TestingPrefServiceSyncable pref_service_;
+  network::TestURLLoaderFactory test_url_loader_factory_;
+  TestSigninClient signin_client_;
+  std::unique_ptr<IdentityManager> identity_manager_;
+  std::unique_ptr<TestIdentityManagerObserver> identity_manager_observer_;
+  CoreAccountId primary_account_id_;
+  GaiaCookieManagerService* gaia_cookie_manager_service_;
+};
+
+TEST_F(BraveIdentityManagerTest, GetAccountsInCookieJarWithNoAccounts) {
+  base::RunLoop run_loop;
+  identity_manager_observer()->SetOnAccountsInCookieUpdatedCallback(
+      run_loop.QuitClosure());
+
+  SetListAccountsResponseNoAccounts(test_url_loader_factory());
+
+  TriggerListAccounts();
+  const AccountsInCookieJarInfo& accounts_in_cookie_jar =
+      identity_manager()->GetAccountsInCookieJar();
+  EXPECT_FALSE(accounts_in_cookie_jar.accounts_are_fresh);
+  EXPECT_TRUE(accounts_in_cookie_jar.signed_in_accounts.empty());
+  EXPECT_TRUE(accounts_in_cookie_jar.signed_out_accounts.empty());
+
+  run_loop.Run();
+
+  TriggerListAccounts();
+  const AccountsInCookieJarInfo updated_accounts_in_cookie_jar =
+      identity_manager()->GetAccountsInCookieJar();
+
+  EXPECT_FALSE(updated_accounts_in_cookie_jar.accounts_are_fresh);
+  EXPECT_TRUE(updated_accounts_in_cookie_jar.signed_in_accounts.empty());
+  EXPECT_TRUE(updated_accounts_in_cookie_jar.signed_out_accounts.empty());
+}
+
+TEST_F(BraveIdentityManagerTest, GetAccountsInCookieJarWithOneAccount) {
+  base::RunLoop run_loop;
+  identity_manager_observer()->SetOnAccountsInCookieUpdatedCallback(
+      run_loop.QuitClosure());
+
+  SetListAccountsResponseOneAccount(kTestEmail, kTestGaiaId,
+                                    test_url_loader_factory());
+
+  TriggerListAccounts();
+  const AccountsInCookieJarInfo& accounts_in_cookie_jar =
+      identity_manager()->GetAccountsInCookieJar();
+  EXPECT_FALSE(accounts_in_cookie_jar.accounts_are_fresh);
+  EXPECT_TRUE(accounts_in_cookie_jar.signed_in_accounts.empty());
+  EXPECT_TRUE(accounts_in_cookie_jar.signed_out_accounts.empty());
+
+  run_loop.Run();
+
+  TriggerListAccounts();
+  const AccountsInCookieJarInfo& updated_accounts_in_cookie_jar =
+      identity_manager()->GetAccountsInCookieJar();
+
+  EXPECT_FALSE(updated_accounts_in_cookie_jar.accounts_are_fresh);
+  ASSERT_EQ(0u, updated_accounts_in_cookie_jar.signed_in_accounts.size());
```

Comment:
Why using `ASSERT` here specifically?  And we should probably just check it with `empty()` method like rest of the tests

Reply:
Thanks. My mistake, I took these lines from the Chromium test, where later was check of each element in `signed_in_accounts` , so at that place `ASSERT` stops test execution before memory violation. In my test it should  be `EXPECT`. Fixed.